keys/keys.go

```diff
@@ -355,3 +355,14 @@ func MetaScanBounds(key proto.Key) (proto.Key, proto.Key) {
 	// Otherwise find the first entry greater than the given key in the same meta prefix.
 	return key.Next(), proto.Key(key[:len(Meta1Prefix)]).PrefixEnd()
 }
+
+// MetaReverseScanBounds returns the start and end keys of the range within which the
+// desired meta record can be found by means of an engine reverse scan. The given key
+// must be a valid RangeMetaKey as defined by ValidateRangeMetaKey.
+func MetaReverseScanBounds(key proto.Key) (proto.Key, proto.Key, error) {
+	if key.Equal(proto.KeyMin) || key.Equal(Meta1Prefix) {
+		return nil, nil, NewInvalidRangeMetaKeyError("KeyMin or Meta1Prefix can't be used as the key of reverse scan", key)
+	}
+	// Otherwise find the first entry less than or equal to the given key in the same meta prefix.
+	return proto.Key(key[:len(Meta1Prefix)]), key, nil
```

Comment:
I don't think the cast to `proto.Key` is necessary here (try removing it). Likely this was copy&pasted from elsewhere in this file which contains other unnecessary casts to `proto.Key` (unnecessary because they are redundant, the type being casted is already a `proto.Key`).